### ISSUE TITLE
How to control version of code in the services

### ISSUE THREAD

**Issue by kant2002** [dotnet/tye#438]
When run `tye --deploy` from tutorial it is creates following output
```
Building Docker Image...
        Created Docker Image: 'xxxx.azurecr.io/frontend:1.0.0'
Pushing Docker Image...
        Pushed docker image: 'xxxx.azurecr.io/frontend:1.0.0'
```

From documentation not clear how that version can be changed. For example I usually control version via `/p:VersionSuffix`, how that can be simulated? Or if I use `Directory.Build.props` in the parent folder, will this be preserved?

**Comment by rynowak** [comment]
We get the version number from MSBuild, specifically the `InformationalVersion` or `Version` property. I've tested this with Nerdbank.GitVersioning in the past. If you're setting version-related properties in MSBuild it should be honored. If not, show my a repro and we'll investigate 👍 

Do you usually pass the `/p:VersionSuffix` at the command-line - rather than storing the version somewhere?

**Comment by kant2002** [comment]
Yes. I do usually pass via `/p:VersionSuffix` and this is passed in CI build usually. I maybe try to adopt `Nerdbank.GitVersioning`, but I really would like to have something 
```
tye deploy --interactive /p:VersionSuffix "ci-10000"
```
What if I want Debug version instead of Release to be deployed (is this true right now?). I can imagine different configurations can be deployed. For example when I have custom build with more verbose logging enabled. I'm happy with documentation fixes/suggestions 😄

**Comment by rynowak** [comment]
Ok thanks, I understand.

We have the ability to pass build args via tye.yaml - but it sounds like there's a need to enable this at the commandline as well.

If you don't mind, I'm going to change the title to reflect that so that it's clearer. Let me know if you think I missed something.

**Comment by kant2002** [comment]
No, you don't miss anything I believe. I'm willing to contribute if this is more or less simple thing to implement.

**Comment by rynowak** [comment]
Cool. Here's what I think would be required:

- Add a new command line switch - it should probably look like `--build-property VersionSuffix=ci-1000`
- In the code for the various commands where we do get back an `ApplicationBuilder`, loop through each project and add this to `BuildProperties` ([here](https://github.com/dotnet/tye/blob/master/src/Microsoft.Tye.Core/ProjectServiceBuilder.cs#L54)])
- Add an E2E test for `tye generate` that matches your scenario

This should be enough to get it through both the evalution step and build step, because that plumbing's already done.

This should work for the following commands:

- Generate
- Push
- Build
- Run
- Deploy

**Comment by jkotalik** [comment]
We should verify what works and doesn't work here.
